subsys/net/lib/sockets/sockets.c

```diff
@@ -891,14 +985,128 @@ static int sock_ioctl_vmeth(void *obj, unsigned int request, ...)
 	case ZFD_IOCTL_CLOSE:
 		return zsock_close_ctx(obj);
 
+	case ZFD_IOCTL_POLL_PREPARE: {
+		va_list args;
+		struct zsock_pollfd *pfd;
+		struct k_poll_event **pev;
+		struct k_poll_event *pev_end;
+
+		va_start(args, request);
+		pfd = va_arg(args, struct zsock_pollfd *);
+		pev = va_arg(args, struct k_poll_event **);
+		pev_end = va_arg(args, struct k_poll_event *);
+		va_end(args);
+
+		return zsock_poll_prepare_ctx(obj, pfd, pev, pev_end);
+	}
+
+	case ZFD_IOCTL_POLL_UPDATE: {
+		va_list args;
+		struct zsock_pollfd *pfd;
+		struct k_poll_event **pev;
+
+		va_start(args, request);
+		pfd = va_arg(args, struct zsock_pollfd *);
+		pev = va_arg(args, struct k_poll_event **);
+		va_end(args);
+
+		return zsock_poll_update_ctx(obj, pfd, pev);
+	}
+
 	default:
 		errno = EOPNOTSUPP;
 		return -1;
 	}
 }
 
-static const struct fd_op_vtable sock_fd_op_vtable = {
-	.read = sock_read_vmeth,
-	.write = sock_write_vmeth,
-	.ioctl = sock_ioctl_vmeth,
+static int sock_close_vmeth(void *obj)
+{
+	return zsock_close_ctx(obj);
+}
+
+static int sock_bind_vmeth(void *obj, const struct sockaddr *addr,
```

Comment:
Ummm, and we need all these wrappers because ... ?

Reply:
Beacuse  for instance `sock_bind_vmeth` and `zsock_bind_ctx` ahve different signatures (`void *obj` vs `struct net_context *ctx`). GCC will throw warning when using `zsock_bind_ctx` in the vtable directly.